agent/agent.go

```diff
@@ -94,6 +101,97 @@ type outputUnit struct {
 	outputs []*models.RunningOutput
 }
 
+func (a *Agent) RunSingleInput(input_config *models.InputConfig, ctx context.Context) {
+	// validating if an input plugin is already running, and therefore shouldn't be run again
+	_, ok := a.inputStatus[input_config.Name]
+	if ok {
+		log.Printf("E! [agent] You are trying to run an input that is already running: %s \n", input_config.Name)
+		return
+	}
+
+	startTime := time.Now()
+
+	// seeing if the plugin exists
+	plugin, ok := inputs.Inputs[input_config.Name]
+	if !ok {
+		log.Printf("E! [agent] input config's name is not valid: %s \n", input_config.Name)
+		return
+	}
+	input := models.NewRunningInput(plugin(), input_config)
+
+	// Overwrite agent interval if this plugin has its own.
+	interval := a.Config.Agent.Interval.Duration
+	if input.Config.Interval != 0 {
+		interval = input.Config.Interval
+	}
+
+	// Overwrite agent precision if this plugin has its own.
+	precision := a.Config.Agent.Precision.Duration
+	if input.Config.Precision != 0 {
+		precision = input.Config.Precision
+	}
+
+	// Overwrite agent collection_jitter if this plugin has its own.
+	jitter := a.Config.Agent.CollectionJitter.Duration
+	if input.Config.CollectionJitter != 0 {
+		jitter = input.Config.CollectionJitter
+	}
+
+	var ticker Ticker
+	if a.Config.Agent.RoundInterval {
+		ticker = NewAlignedTicker(startTime, interval, jitter)
```

Comment:
I think AlignedTicker might actually be trying to align all the metrics with one universal startTime, so we might need to save the original startTime from ```runInputs``` and then reuse that? Not sure though, maybe you guys can look into it. I'm just guessing right now.